lib/items.js

```diff
@@ -0,0 +1,82 @@
+/*!
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+const jsonmergepatch = require("json-merge-patch"),
+      joi = require("joi"),
+      UUID = require("uuid");
+      
+const DataStoreError = require("./util/errors");
+
+const STRING_500 = joi.string().max(500).allow("").allow(null).default("");
+const STRING_10K = joi.string().max(10000).allow("").allow(null).default("");
+const BASE_ENTRY_SCHEMA = joi.object().keys({
+  kind: joi.string().required(),
+  notes: STRING_10K
+});
+const ENTRY_SCHEMAS = [
+  BASE_ENTRY_SCHEMA.keys({
+    kind: "login",
+    username: STRING_500,
+    password: STRING_500
+  }),
+];
+const SCHEMA = joi.object().keys({
+  title: STRING_500,
+  origins: joi.array().items(STRING_500).max(5).default([]),
+  tags: joi.array().items(STRING_500).max(10).default([]),
+  entry: joi.alternatives(ENTRY_SCHEMAS).required(),
+});
+const VALIDATE_OPTIONS = {
+  abortEarly: false,
+  stripUnknown: true
+};
+
+const HISTORY_MAX = 100;
```

Comment:
This worries me a little. Maybe it's just because I'm used to version control, but the idea of history going away after enough revisions makes me nervous. If this is the only/best way to ensure sanity in the datastore, then it's ok though...

Reply:
We won't have infinite storage in the cloud.  I can be talked into a somewhat bigger limit, but I cannot be talked into no limit.